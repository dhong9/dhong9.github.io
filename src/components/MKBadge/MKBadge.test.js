/**
=========================================================
* Danyo-1.2
=========================================================

* Copyright 2024 Danyo (https://www.danyo.tech)

Coded by www.danyo.tech

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

// React testing libraries
import { render } from "@testing-library/react";

// Material Kit 2 React themes
import { ThemeProvider } from "@mui/material/styles";
import theme from "assets/theme";

// Component to test
import MKBadge from "components/MKBadge";

describe("MKBadge", () => {
  it("renders small light badges", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <MKBadge color="light" size="small"></MKBadge>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("renders medium white badges", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <MKBadge color="white" size="medium"></MKBadge>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
