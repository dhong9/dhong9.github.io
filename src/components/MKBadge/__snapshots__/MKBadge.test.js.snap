// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MKBadge renders medium white badges 1`] = `
<div>
  <span
    class="MuiBadge-root css-8u7g6p-MuiBadge-root"
  >
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-14zn51v-MuiBadge-badge"
    />
  </span>
</div>
`;

exports[`MKBadge renders small light badges 1`] = `
<div>
  <span
    class="MuiBadge-root css-kpl07s-MuiBadge-root"
  >
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-14zn51v-MuiBadge-badge"
    />
  </span>
</div>
`;
